src/preload.js

```diff
@@ -21,9 +23,43 @@ function init() {
   window.SerialPort = SerialPort;
   window.MakerBridge = {
     getVersion,
+    flashBoardFirmware,
   };
 }
 
+/**
+ * Flash firmware of board with contents of specified hex file.
+ * @param {Object} flashInfo, info to flash board
+ * @param {String} flashInfo.boardName, name of board from Avrgirl options
+ * @param {String} flashInfo.hexPath, path to hex file to flash to board
+ * @param {String} flashInfo.checksum, checksum to verify valid hex download
+ * @return {Promise}
+ */
+function flashBoardFirmware(flashInfo) {
+  return new Promise((resolve, reject) => {
```

Comment:
Does the Electron runtime not support or transpile async/await?

Reply:
I'm not sure about Electron, but the code that uses this function in the CDO repo uses promises, so for consistency, I went with that. Would it be stylistically okay to have this be async/await while the code that calls on it relies on promises?

Reply:
Yep! An async function is just syntax sugar for a function that [always returns a promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function). You can use them interchangeably!